plugins/plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/DockerContainerNameGeneratorTest.java

```diff
@@ -0,0 +1,177 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.machine;
+
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.commons.lang.Pair;
+import org.eclipse.che.commons.user.UserImpl;
+import org.mockito.InjectMocks;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static java.util.Collections.emptyList;
+import static org.eclipse.che.plugin.docker.machine.DockerContainerNameGenerator.ContainerNameInfo;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
+
+/**
+ * Test for {@link DockerContainerNameGenerator}
+ *
+ * @author Alexander Andrienko
+ */
+@Listeners(MockitoTestNGListener.class)
+public class DockerContainerNameGeneratorTest {
+    private static final String WORKSPACE_ID = "workspacebbbx2ree3iykn8gc";
+    private static final String MACHINE_NAME = "ws-machine";
+    private static final String MACHINE_ID   = "machineic131ppamujngv6y";
+    private static final String USER_NAME    = "some_user";
+
+    @InjectMocks
+    private DockerContainerNameGenerator nameGenerator;
+
+    @BeforeMethod
+    public void setUp() {
+        EnvironmentContext.getCurrent().setUser(new UserImpl(USER_NAME, "stub", "stub", emptyList(), false));
+    }
+
+    @Test
+    public void containerNameShouldBeGenerated() {
+        String expectedResult = "workspacebbbx2ree3iykn8gc_machineic131ppamujngv6y_some_user_ws-machine";
```

Comment:
Please add tests for usernames, machine names with invalid characters and upper cased characters